server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -1106,4 +1100,81 @@ public void unregisterDynamicAction(ActionType<?> action) {
             return registry.get(action);
         }
     }
+
+    /**
+     * The DynamicRouteRegistry maintains a registry mapping {@link org.opensearch.rest.RestHandler.Route} instances to {@link org.opensearch.rest.extensions.RestSendToExtensionAction} instances.
+     * <p>
+     * This class is modeled after {@link NamedRegistry} but provides both register and unregister capabilities.
+     *
+     * @opensearch.internal
+     */
+    public static class DynamicRouteRegistry {
```

Comment:
Just thinking out loud. Do you think the right place for this class should be in `RestHandler`?

Reply:
See you have removed the class.